### ISSUE TITLE
Webhook filter with invalid event type or event handler does not result in error or warning to users

### ISSUE THREAD

**Issue by lichenyang2** [couchbase/sync_gateway#661]
In 1.1.0, the only supported event type is "document_changed".  When I specify "document_added" or "zzzzz" as event type, no event is posted, but also there is error or warning message displayed in the gateway log to let user know there is an error in the event section of the gateway.config.

Example of bad event type:

```
{
    "adminInterface": ":4985",
    "interface": ":4984",
    "log": ["HTTP", "HTTP+", "CRUD", "Events+"],
    "databases": {
        "db": {
            "server": "walrus:",
            "users": {
                 "GUEST": {"disabled": false, "admin_channels": ["*"] }
            },
            "event_handlers": {
                "document_added": [
                    {"handler": "webhook",
                     "url": "http://localhost:8081",
                     "filter": `function(doc) {
                        if (doc.class == "English" && doc.grade == "A" && doc.count > 3 && doc.bool == true) {
                            return true;
                        }
                        return false;
                     }`
                    } 
                ]
            }
        }    
    }
} 
```

Example of bad event handler:

```
{
    "adminInterface": ":4985",
    "interface": ":4984",
    "log": ["HTTP", "HTTP+", "CRUD", "Events+"],
    "databases": {
        "db": {
            "server": "walrus:",
            "users": {
                 "GUEST": {"disabled": false, "admin_channels": ["*"] }
            },
            "event_handlerssssss": {
                "document_changed": [
                    {"handler": "webhook",
                     "url": "http://localhost:8081",
                     "filter": `function(doc) {
                        if (doc.class == "English" && doc.grade == "A" && doc.count > 3 && doc.bool == true) {
                            return true;
                        }
                        return false;
                     }`
                    } 
                ]
            }
        }    
    }
} 
```

Example output of gateway.log (it shows no event is sent):

```
17:03:47.148900 Enabling logging: [HTTP HTTP+ CRUD Events+]
17:03:47.149080 ==== Couchbase Sync Gateway/0.0.0(345;5385fdc) ====
17:03:47.149170 Opening db /db as bucket "db", pool "default", server <walrus:>
17:03:47.149328 Opening Walrus database db on <walrus:>
17:03:47.150567 Using default sync function 'channel(doc.channels)' for database "db"
17:03:47.150922     Reset guest user to config
17:03:47.150994 Starting admin server on :4985
17:03:47.161466 Starting server on :4984 ...
17:03:47.720650 HTTP:  #001: PUT /db/doc45  (ADMIN)
17:03:47.721802 CRUD: Stored doc "doc45" / "1-e72e07f47d60617e140e9f90d51c691c"
17:03:47.721995 HTTP+: #001:     --> 201   (1.4 ms)
```

**Comment by zgramana** [comment]
Config errors, e.g. invalid key names, should be output when parsing the config during process start. This should be addressed by the 1.1 GA, as it's a core usability issue.

**Comment by adamcfraser** [comment]
Validation of property values would be straightforward, but in this case we're talking about trying to validate invalid key names.  This is a non-trivial enhancement, as we're using the standard go JSON unmarshal to parse the config, which intentionally doesn't do any schema-type validation (i.e. it's looking for existence of named properties, and ignores anything it doesn't care about.

We could consider schema enforcement for our configs (which I think is where #826 might end up), but I feel like it may be a high-risk change for 1.1 at this point in the release.  

I'm still looking into whether there's a workaround that we can apply specifically to the `event_handlers` sub-document processing, but I'm not convinced it's worth the effort before we tackle #826.
